cmd/create/create_agent.go

```diff
@@ -0,0 +1,439 @@
+package create
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/armory/armory-cli/pkg/config"
+	"github.com/armory/armory-cli/pkg/configuration"
+	"github.com/armory/armory-cli/pkg/model"
+	"github.com/armory/armory-cli/pkg/util"
+	"github.com/manifoldco/promptui"
+	"github.com/samber/lo"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/cmd/apply"
+	"k8s.io/kubectl/pkg/cmd/delete"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"time"
+)
+
+const (
+	agentShort   = "Create an agent"
+	agentLong    = "Create an agent"
+	agentExample = `
+	  # Create a new agent
+	  armory create agent`
+
+	defaultNamespaceName = "armory-rna"
+	defaultSecretName    = "rna-client-credentials"
+)
+
+var agentConnectedPollRate = time.Second * 10
+
+type AgentOptions struct {
+	// Name of resource being created
+	Name      string
+	Namespace string
+
+	Context context.Context
+
+	ArmoryClient      *configuration.ConfigClient
+	configuration     *config.Configuration
+	contextNames      []string
+	credentials       *model.Credential
+	kubernetesFactory cmdutil.Factory
+	configAccess      clientcmd.ConfigAccess
+	KubernetesClient  corev1client.CoreV1Interface
+}
+
+// NewAgentOptions creates a new *AgentOptions with sane defaults
+func NewAgentOptions() *AgentOptions {
```

Comment:
I would say that this isn't very go idiomatic, especially since you're not setting any non-default values. I also would not expose this as a public function since it doesn't look like you're using it anywhere outside of the package.